packages/worker/src/api/controllers/global/users.ts

```diff
@@ -22,6 +22,65 @@ export const save = async (ctx: any) => {
   }
 }
 
+export const bulkSave = async (ctx: any) => {
+  let { users: newUsersRequested, groups } = ctx.request.body
+  let groupsToSave: any[] = []
+  const newUsers: any[] = []
+  const db = tenancy.getGlobalDB()
+  const currentUserEmails =
+    (await users.allUsers())?.map((x: any) => x.email) || []
+  for (const newUser of newUsersRequested) {
+    if (
+      newUsers.find((x: any) => x.email === newUser.email) ||
+      currentUserEmails.includes(newUser.email)
+    )
+      continue
+    newUser.userGroups = groups
+    newUsers.push(newUser)
+  }
+
+  if (groups.length) {
+    groups.forEach(async (groupId: string) => {
+      let oldGroup = await db.get(groupId)
+      groupsToSave.push(oldGroup)
+    })
+  }
+
+  try {
+    let response: any[] = []
+    for (const user of newUsers) {
+      response.push(
+        await users.save(user, {
+          hashPassword: true,
+          requirePassword: user.requirePassword,
+          bulkCreate: false,
+        })
+      )
+    }
+
+    // delete passwords and add to group
+    newUsers.forEach(user => {
+      delete user.password
+    })
+
+    if (groupsToSave.length) {
+      groupsToSave.forEach(async (userGroup: UserGroup) => {
+        userGroup.users = [...userGroup.users, ...response]
+        await db.put(userGroup)
```

Comment:
Same as above, this one is less important as I assume there couldn't be that many groups, but when you are doing a raw `db.put` in a loop of some sort you could extract it to an array and then perform a `db.bulkDocs` after the loop for efficiency reasons.